src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -252,6 +410,11 @@ public static List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinitio
                         continue;
                     }
 
+                    if (compilerGeneratedBranchesToExclude.Contains(instruction.Offset))
```

Comment:
Is it too complex avoid to generate the list above but check directly when we come to the branch instruction to exclude?

Reply:
To generate this list we need to parse all method instructions, so I prefer to do that before the foreach (I thought that in the future we may have other branches to load in that list). If you prefer, I can move the code in a Skip style method and cache the list.. something like:
```csharp
SkipCompilerGeneratedBranchesForExceptionHandlersInsideAsyncStateMachine(Instruction instruction)
{

    if (CompilerGeneratedBranchesToExclude == null)
    {
         // Detect branches to exclude
    }

    return compilerGeneratedBranchesToExclude.Contains(instruction.Offset);
}
```

Reply:
Yep better

Reply:
Not so better maybe. The list is for method, so is not useful to use static cache.  How can I refactor this code?

Reply:
@MarcoRossignoli please check last update